**README:**
- Topic: How excessive use of digital devices can lead to eating disorders
- Look at important variables:
  - Test

**Prepared by:**
- Email: wiener baalls

# Exploratory Data Analysis

This is an accompanying notebook for the Exploratory Data Analysis slide deck.

In [2]:
# @title Import needed libraries
import pandas as pd
import numpy as np

# @title Import Google Drive directory
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
# @title Import DataFrame
file_path = '/content/drive/MyDrive/Colab Notebooks/colab-files/eating-disorders-bangladeshi-uni.csv'
df = pd.read_csv(file_path)

In [4]:
# @title View a preview of the DataFrame
df.head(10)

Age_Range  Gender Division_ofOrigin      Current_EducationLevel  \
0     18-24    Male          Rajshahi  Bachelor's degree 3rd year   
1     25-34  Female        Chattogram    Master's degree 2nd year   
2     18-24    Male            Khulna  Bachelor's degree 2nd year   
3     18-24  Female             Dhaka  Bachelor's degree 2nd year   
4     18-24    Male            Sylhet  Bachelor's degree 3rd year   
5     18-24  Female            Khulna  Bachelor's degree 2nd year   
6     18-24    Male          Barishal  Bachelor's degree 3rd year   
7     18-24    Male            Sylhet  Bachelor's degree 1st year   
8     18-24    Male            Sylhet  Bachelor's degree 3rd year   
9     25-34    Male          Rajshahi  Bachelor's degree 3rd year   

    Employment_Status     Marital_Status EatLess_OnWeightGain  \
0             Student  In a relationship                Often   
1       Self-employed            Married                Often   
2             Student             Single                Often   
3       Self-employed            Married                Often   
4             Student             Single            Sometimes   
5       Self-employed            Married                Often   
6             Student            Married                Often   
7  Employed full-time             Single                Never   
8       Self-employed             Single                Often   
9             Student             Single                Often   

  EatLess_AtMealtime RefuseFood_WeightConcern Monitor_Food  ...  \
0              Never                Sometimes    Sometimes  ...   
1              Often                   Seldom        Often  ...   
2              Never                    Often       Seldom  ...   
3              Often                    Never    Sometimes  ...   
4              Never                   Seldom        Often  ...   
5              Often                    Often    Sometimes  ...   
6              Often                    Never       Seldom  ...   
7             Seldom                   Seldom    Sometimes  ...   
8              Often                   Seldom        Never  ...   
9             Seldom                Sometimes        Often  ...   

  Days_TriedLimitFoodToControlShapeOrWeight Days_FastedToControlShapeOrWeight  \
0                                13-27 days                         Every day   
1                                 1-12 days                        13-27 days   
2                                 Every day                         1-12 days   
3                                 Every day                         1-12 days   
4                                 1-12 days                        13-27 days   
5                                13-27 days                        13-27 days   
6                                13-27 days                           No days   
7                                   No days                         1-12 days   
8                                   No days                         Every day   
9                                 1-12 days                         1-12 days   

  Days_ExcludedFoodToControlShapeOrWeight  \
0                               1-12 days   
1                              13-27 days   
2                               1-12 days   
3                               Every day   
4                               1-12 days   
5                                 No days   
6                              13-27 days   
7                               Every day   
8                                 No days   
9                               Every day   

  Days_FollowedRulesToControlShapeOrWeight Days_FearLosingControlOverEating  \
0                                Every day                        1-12 days   
1                                  No days                       13-27 days   
2                               13-27 days                        1-12 days   
3                                1-12 days                        1-12 days   
4    

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 550 entries, 0 to 549
Data columns (total 57 columns):
 #   Column                                         Non-Null Count  Dtype 
---  ------                                         --------------  ----- 
 0   Age_Range                                      550 non-null    object
 1   Gender                                         550 non-null    object
 2   Division_ofOrigin                              550 non-null    object
 3   Current_EducationLevel                         550 non-null    object
 4   Employment_Status                              550 non-null    object
 5   Marital_Status                                 550 non-null    object
 6   EatLess_OnWeightGain                           550 non-null    object
 7   EatLess_AtMealtime                             550 non-null    object
 8   RefuseFood_WeightConcern                       550 non-null    object
 9   Monitor_Food                                   550 non-null    ob

## 1 Measures of Central Tendency

In [ ]:
# @title Create sample DataFrames

df1 = pd.DataFrame({"X": [54, 54, 54, 55, 56, 57, 57, 58, 58, 60, 60]})
df2 = pd.DataFrame({"X": [1, 2, 54, 54, 54, 55, 56, 57, 57, 58, 58, 60, 60, 200, 250]})

In [ ]:
# @title Mean, Median, Mode

mean1 = df1["X"].mean().round(1)
median1 = df1["X"].median()
mode1 = df1["X"].mode()[0]

print(f"[DF1] mean = {mean1}, median = {median1}, mode = {mode1}")

mean2 = df2["X"].mean().round(1)
median2 = df2["X"].median()
mode2 = df2["X"].mode()[0]

print(f"[DF2] mean = {mean2}, median = {median2}, mode = {mode2}")

[DF1] mean = 56.6, median = 57.0, mode = 54
[DF2] mean = 71.7, median = 57.0, mode = 54


In [ ]:
# @title Trimmed Mean

from scipy.stats import trim_mean

trimmed_mean2 = trim_mean(df2["X"], proportiontocut=0.15).round(1)

print(f"[DF2] trimmed mean (removed 15%) = {trimmed_mean2}")

[DF2] trimmed mean (removed 15%) = 56.6


In [4]:
?trim_mean


# penois

Object `trim_mean` not found.


## 2 Measures of Dispersion

In [ ]:
# @title Create sample DataFrames

df1 = pd.DataFrame({"X": [54, 54, 54, 55, 56, 57, 57, 58, 58, 60, 60]})
df2 = pd.DataFrame({"X": [1, 2, 54, 54, 54, 55, 56, 57, 57, 58, 58, 60, 60, 200, 250]})

In [ ]:
# @title Range

range1 = df1["X"].max() - df1["X"].min()
print(f"[DF1] range = {range1}")

range2 = df2["X"].max() - df2["X"].min()
print(f"[DF2] range = {range2}")

[DF1] range = 6
[DF2] range = 249


In [ ]:
# @title Interquartile Range

q1_1 = df1["X"].quantile(0.25)
q3_1 = df1["X"].quantile(0.75)
iqr_1 = q3_1 - q1_1
print(f"[DF1] IQR = {q3_1} - {q1_1} = {iqr_1}")

q1_2 = df2["X"].quantile(0.25)
q3_2 = df2["X"].quantile(0.75)
iqr_2 = q3_2 - q1_2
print(f"[DF2] IQR = {q3_2} - {q1_2} = {iqr_2}")

[DF1] IQR = 58.0 - 54.5 = 3.5
[DF2] IQR = 59.0 - 54.0 = 5.0


The IQR in the notebook differs from the ones in the slide due to how pandas computes for the quantile. See the `interpolation` parameter.

In [ ]:
?df1.quantile

In [ ]:
# @title Variance and Standard Deviation

mean1 = df1["X"].mean()
variance1 = df1["X"].var()
std1 = df1["X"].std()
print(f"[DF1] mean = {mean1:.2f}, variance = {variance1:.2f}, standard deviation = {std1:.2f}")

mean2 = df2["X"].mean()
variance2 = df2["X"].var()
std2 = df2["X"].std()
print(f"[DF2] mean = {mean2:.2f}, variance = {variance2:.2f}, standard deviation = {std2:.2f}")

[DF1] mean = 56.64, variance = 5.05, standard deviation = 2.25
[DF2] mean = 71.73, variance = 4332.50, standard deviation = 65.82


DF1 Interpretation
- The data points in DF1 are tightly clustered around the mean.
- The standard deviation of 2.25 means most values fall roughly within ±2.25 units of the mean (i.e., between about 54.4 and 58.9).
- Variance is low, which implies that DF1's data is more reliable/stable.

DF2 Interpretation
- DF2's data points are much more spread out around the mean.
- A standard deviation of 65.82 means values vary widely, typically within ±65.82 units of the mean (so roughly between 6 and 138).
- The very high variance confirms there is huge variability or inconsistency in DF2's data.
- DF2's data shows greater diversity or possibly the presence of outliers/extreme values

## 3 Correlation

In [ ]:
# @title Create sample DataFrames

data = {
    "height": [154, 160, 167, 152, 175, 161, 168],
    "weight": [60, 70, 65, 55, 85, 80, 75]
}

df = pd.DataFrame(data)
df

height  weight
0     154      60
1     160      70
2     167      65
3     152      55
4     175      85
5     161      80
6     168      75

### 3.1 Pearson Correlation

$$r_{xy} = \frac{cov(x,y)}{\sigma_x \cdot  \sigma_y} = \frac{\sum_{i=1}^{n}(x_i-\overline{x})(y_i-\overline{y})}{\sqrt{\sum_{i=1}^{n}(x_i-\overline{x})^2} \cdot \sqrt{\sum_{i=1}^{n}(y_i-\overline{y})^2}}$$

In [ ]:
# @title Manual Computaion: Expand DataFrame

df["x_minus_mean"] = df["height"] - df["height"].mean()
df["y_minus_mean"] = df["weight"] - df["weight"].mean()
df["product"] = df["x_minus_mean"] * df["y_minus_mean"]
df["x_minus_mean_squared"] = df["x_minus_mean"] ** 2
df["y_minus_mean_squared"] = df["y_minus_mean"] ** 2
df

height  weight  x_minus_mean  y_minus_mean     product  \
0     154      60     -8.428571         -10.0   84.285714   
1     160      70     -2.428571           0.0   -0.000000   
2     167      65      4.571429          -5.0  -22.857143   
3     152      55    -10.428571         -15.0  156.428571   
4     175      85     12.571429          15.0  188.571429   
5     161      80     -1.428571          10.0  -14.285714   
6     168      75      5.571429           5.0   27.857143   

   x_minus_mean_squared  y_minus_mean_squared  
0             71.040816                 100.0  
1              5.897959                   0.0  
2             20.897959                  25.0  
3            108.755102                 225.0  
4            158.040816                 225.0  
5              2.040816                 100.0  
6             31.040816                  25.0

In [ ]:
# @title Manual Computation: Compute for Pearson Correlation

numerator = df["product"].sum()
denominator = np.sqrt(df["x_minus_mean_squared"].sum()) * np.sqrt(df["y_minus_mean_squared"].sum())
pearson = numerator / denominator

print(f"r = {pearson:.3f}")

r = 0.796


In [ ]:
# @title Automatic Computation: Using pandas `corr()` function

pearson = df["height"].corr(df["weight"], method="pearson")
print(f"r = {pearson:.3f}")

r = 0.796


### 3.2 Spearman Correlation

$$r_{xy} = \frac{cov(rank(x), rank(y))}{\sigma_{rank(x)} \cdot  \sigma_{rank(y)}} = 1 - \frac{6\sum{d_i^2}}{n(n^2-1)}$$

In [ ]:
# @title Manual Computaion: Expand DataFrame

df["rank_x"] = df["height"].rank()
df["rank_y"] = df["weight"].rank()
df["d"] = df["rank_x"] - df["rank_y"]
df["d_squared"] = df["d"] ** 2

df

height  weight  rank_x  rank_y    d  d_squared
0     154      60     2.0     2.0  0.0        0.0
1     160      70     3.0     4.0 -1.0        1.0
2     167      65     5.0     3.0  2.0        4.0
3     152      55     1.0     1.0  0.0        0.0
4     175      85     7.0     7.0  0.0        0.0
5     161      80     4.0     6.0 -2.0        4.0
6     168      75     6.0     5.0  1.0        1.0

In [ ]:
# @title Manual Computation: Compute for Spearman Correlation

numerator = 6 * df["d_squared"].sum()
denominator = len(df) * (len(df) ** 2 - 1)
spearman = 1 - (numerator / denominator)

print(f"r = {spearman:.3f}")

r = 0.821


In [ ]:
# @title Automatic Computation: Using pandas `corr()` function

spearman = df["height"].corr(df["weight"], method="spearman")
print(f"r = {spearman:.3f}")

r = 0.821
